crates/common/download/src/download.rs

```diff
@@ -408,6 +391,19 @@ impl Downloader {
     }
 }
 
+/// Decides whether HTTP request error is retryable.
+fn reqwest_err_to_backoff(err: reqwest::Error) -> backoff::Error<reqwest::Error> {
+    if err.is_timeout() || err.is_connect() {
+        return backoff::Error::transient(err);
+    }
+    if let Some(status) = err.status() {
+        if status.is_server_error() {
```

Comment:
What about 404? In most cases it might be non-retryable, but aren't there cases where a resource not existing also could be transient (delays in getting a newly uploaded resource synced across all regions or something like that)?

Reply:
Yeah, currently 404 is considered permanent. And perhaps indeed we could retry on 404 for such scenarios, but then we add delay in cases where the URL is simply wrong.
I have no preference either way. @reubenmiller what do you think?

Reply:
I would consider 404 as a permanent error.

To handle the case where a resource is not existing *yet* but which creation is pending,  the solution will depends on the server but should not rely on 404, but using a combination of 202 Accepted (the creation of the resource has been accepted) and 400 Bad Request (possibly with a hint to retry in a sec).